src/Razor/src/Microsoft.AspNetCore.Razor.LanguageServer/Semantic/Services/DefaultRazorSemanticTokensInfoService.cs

```diff
@@ -318,38 +180,38 @@ internal class DefaultRazorSemanticTokensInfoService : RazorSemanticTokensInfoSe
         }
 
         // Internal and virtual for testing only
-        internal virtual async Task<VersionedSemanticRange> GetCSharpSemanticRangesAsync(
+        internal virtual async Task<IReadOnlyList<SemanticRange>?> GetCSharpSemanticRangesAsync(
             RazorCodeDocument codeDocument,
             TextDocumentIdentifier textDocumentIdentifier,
-            Range? range,
+            Range range,
             long documentVersion,
             CancellationToken cancellationToken,
             string? previousResultId = null)
         {
             var razorRanges = new List<SemanticRange>();
 
-            if (!TryGetMinimalCSharpRange(codeDocument, out var csharpRange))
+            if (!_documentMappingService.TryMapToProjectedDocumentRange(codeDocument, range, out var csharpRange))
```

Comment:
Ohh, I like this change! Might want to specify the parameter as `razorRange` just for clarity though.